GoogleForCreators/web-stories-wp
# Undoing a layer rename reverts to rename when going back to edit mode.

## Context

After renaming a layer and undoing the layer name the layer name uses the wrong value when re-entering edit mode.

See:
https://github.com/GoogleForCreators/web-stories-wp/pull/11676#issuecomment-1153807461



## Summary

I think the issue here is the useState + initial value

> the initial value matters only the first time a component renders. After that, it will preserve the state between renders.

https://dev.to/cronokirby/react-pitfalls-usestate-initialization-39be

This PR adds a useEffect to ensure the "initial value" element.layerName gets updates



## Relevant Technical Choices



## To-do



## User-facing changes



## Testing Instructions






This PR can be tested by following these steps:
1. Enable the layer renaming feature
2. Add a new element to the stage
3. Open the layers panel
4. Rename the layer
5. Undo the rename 
6. Enter edit mode to rename the layer again --- the value should match the current layer name


## Reviews

### Does this PR have a security-related impact?

No



### Does this PR change what data or activity we track or use?

No



### Does this PR have a legal-related impact?

No



## Checklist




---



Fixes #11686
